stellargraph/mapper/sampled_node_generators.py

```diff
@@ -495,6 +496,14 @@ def sample_features(self, head_nodes, batch_num):
 
         return batch_feats
 
+    def default_corrupt_input_index_groups(self):
+        # every sample of a given node type can be grouped together
+        indices_per_nt = defaultdict(list)
+        for tensor_idx, (nt, _) in enumerate(self._sampling_schema[0]):
+            indices_per_nt[nt].append(tensor_idx)
+
+        return sorted(sorted(idx) for idx in indices_per_nt.values())
```

Comment:
I'm not sure but I think this list of lists is already sorted.

 ```suggestion
        return list(indices_per_nt.values())
```

It also seems like the order of both the groups and the elements within the groups are irrelevant anyway.

Reply:
I agree it's irrelevant semantically, but it does change the final result (if the order of these elements changes, the order of the concatenation will change in `CorruptedSequence`, and so the final shuffled result will be different, even if seeds are set). Hence I wanted to have a "canonical" version, so there's no randomness or anything that might make things non-reproducible. 

I also think it's already sorted, with our current implementation of `_sampling_schema`/`GraphSchema.sampling_layout` but I thought this was a low effort and low cost way to ensure that it always stays sorted. I guess another option would be to just have it in a test, but that's somewhat less reliable in my mind.

Reply:
I added comment too, to try to explain/remind for future reference.